parsons/google/google_bigquery.py

```diff
@@ -0,0 +1,89 @@
+from google.cloud.bigquery import Client
+from parsons import Table
+from parsons.google.utitities import setup_google_application_credentials
+from parsons.utilities.files import create_temp_file
+import petl
+import pickle
+
+
+class BigQuery:
+    """
+    Class for querying BigQuery table and returning the data as Parsons tables.
+
+    This class requires application credentials in the form of a json. It can be passed
+    in the following ways:
+
+    * Set an environmental variable named ``GOOGLE_APPLICATION_CREDENTIALS`` with the
+      local path to the credentials json.
+
+      Example: ``GOOGLE_APPLICATION_CREDENTALS='path/to/creds.json'``
+
+    * Pass in the path to the credentials using the ``app_creds`` argument.
+
+    * Pass in a json string using the ``app_creds`` argument.
+
+    Args:
+        project_id: str
+            The project which the client is acting on behalf of. If not passed
+            then will use the default inferred environment.
+        app_creds: str
+            A credentials json string or a path to a json file. Not required
+            if ``GOOGLE_APPLICATION_CREDENTIALS`` env variable set.
+        location: str
+            (optional) Default geographic location for tables
+        client_class: obj
+            (optional) Class to use use the BigQuery client
```

Comment:
This is my stab at trying to make it easy to unit test this class. The idea is that the class to use as the BigQuery client can be passed in, so that in unit testing we can swap out our BigQuery client from `google.cloud.bigquery.client.Client` with a mock so that we aren't actually calling out over the wire.

This dependency injection approach is a little more standard in statically typed languages like Java and C#. I actually think I have a better idea for how to do this without having to have this strange parameter in the class's `__init__`. I will push a commit later today.